rllib/utils/torch_utils.py

```diff
@@ -151,24 +151,27 @@ def convert_to_torch_tensor(x: TensorStructType, device: Optional[str] = None):
         to torch tensors.
 
     Returns:
-        Any: A new struct with the same structure as `stats`, but with all
+        Any: A new struct with the same structure as `x`, but with all
             values converted to torch Tensor types.
     """
 
     def mapping(item):
         if item is None:
             # returns None with dtype=np.obj
             return np.asarray(item)
-        # Already torch tensor -> make sure it's on right device.
-        if torch.is_tensor(item):
-            return item if device is None else item.to(device)
+
         # Special handling of "Repeated" values.
-        elif isinstance(item, RepeatedValues):
+        if isinstance(item, RepeatedValues):
             return RepeatedValues(
                 tree.map_structure(mapping, item.values), item.lengths, item.max_len
             )
+
+        tensor = None
+        # Already torch tensor -> make sure it's on right device.
+        if torch.is_tensor(item):
+            tensor = item
         # Numpy arrays.
-        if isinstance(item, np.ndarray):
```

Comment:
what were we doing before ...  didn't we check these conditions before we get in here :)

Reply:
the else is added because I moved a bit of logic to clean the code. e.g torch.is_tensor() is brought down after dealing with RepeatedValues type.